nntrainer/tensor/tensor_v2.cpp

```diff
@@ -90,6 +90,23 @@ TensorV2::TensorV2(
 }
 #endif
 
+bool TensorV2::operator==(const TensorV2 &rhs) const {
+  /// compares tensor information
+  if (*itensor == *rhs.itensor) {
+    /// compares tensor data
+    if (getDataType() == Tdatatype::FP32) {
+      return *dynamic_cast<FloatTensor *>(itensor) ==
+             *dynamic_cast<FloatTensor *>(rhs.itensor);
+    } else if (getDataType() == Tdatatype::FP16) {
+#ifdef ENABLE_FP16
```

Comment:
it needs to generate exception when datatype is not supported.

Reply:
This function will never be reached if enable-fp16 is not enabled since HalfTensor will never be created in the first place.
Therefore, I removed `throw std::invalid_argument("Error: enable-fp16 is not enabled");`
would it be better to throw an exception with a message stating this should not be reached?

Reply:
exception means it should not happen.
As you mentioned, the case should not be happened, but a developer can make mistakes while maintaining.

Reply:
gotcha. I'll add an exception with a detailed msg. thanks 👍